src/main/scala/org/broadinstitute/dsde/workbench/avram/api/AvramRoutes.scala

```diff
@@ -26,6 +28,22 @@ class AvramRoutes {
     Pong()
   }
 
+  @ApiMethod(name = "authPing", httpMethod = "get", path = "authPing")
+  def authPing(r: HttpServletRequest): Pong = {
+    getToken(r) match {
```

Comment:
Ultimately, I want to make the "routes" layer extremely thin and handle token checks pretty much automatically. I learned some tricks in Scala with Cats that we might be able to use to make the code not even compile if we forget to handle auth (which will hopefully make Bernick happy).

Reply:
Cool -- what kind of tricks are you thinking? Since auth will be common to every API it'd be nice to encapsulate it into some kind of helper function.

I'm just spitballing here:
```
def extractUserInfo(r: HttpServletRequest): Either[ErrorResponse, UserInfo] = ???

@ApiMethod(...)
def authPing(r: HttpServletRequest): Pong = {
  val responseOrError = extractUserInfo(r).map { userInfo =>
    log.info(userInfo)
    Pong()
  }
  // maybe this "get or throw" fold is common enough to make a helper function too
  responseOrError.fold {
    error => throw error.exception,
    result => result
  }
}
```